TileRarities.lua

```diff
@@ -0,0 +1,131 @@
+--[[
+    The data for the different chances and multiplier values can be found on Data:Tile_Chances
+]]--
+local p = {}
+
+function p.main(frame)
+	return p._main(frame.args)
+end
+
+function p._main(args)
+	local json = mw.loadJsonData('Data:Tile_Chances')
+    local ranks = json.ranks
+	local dom = mw.html.create()
+	local tbl = dom:tag('table'):addClass('wikitable'):css('text-align', 'right')
+	local row = tbl:tag('tr')
+	row:tag('th'):attr('rowspan', 2):wikitext('Rarity'):css('vertical-align', 'bottom')
+    for i, v in ipairs(ranks) do
+        row:tag('th'):attr('colspan', 3):wikitext(v.display)
+    end
+    row = tbl:tag('tr')
+    for i, v in ipairs(ranks) do
+        row:tag('th'):wikitext('Chance')
+        row:tag('th'):wikitext('Tile Multiplier')
+        row:tag('th'):wikitext('Gold Multiplier')
+    end
+	for i, v in ipairs(json.chances) do
+		local row = tbl:tag('tr')
+		row:tag('th'):wikitext(v.rarity):css('text-align', 'left')
+        for i2, v2 in ipairs(ranks) do
+            local rarity = v[v2.rank]
+            if rarity ~= nil then
+            	if rarity.chance == nil then
+            		row:tag('td'):wikitext('N/A')
+            	elseif rarity.chance < 0 then
+            		row:tag('td'):wikitext('-')
+        		else
+        			row:tag('td'):wikitext(rarity.chance .. '%')
+    			end
+                row:tag('td'):wikitext(rarity.tile_multiplier)
+                row:tag('td'):wikitext(rarity.gold_multiplier)
+            end
+        end
+	end
+	return dom
+end
+
+function p.wednesday(frame)
```

Comment:
i am not going to suggest to refactor this *now* but youll want to eventually consider refactoring this into a single main function that calls one of `h.printRow`, `h.printWednesdayRow`, or `h.printSundayRow` depending on a parameter called tableType or some such

Reply:
(do that when you feel confident in making the refactor cleaner than the current code, and not before)